lib/App/Sqitch/Engine.pm

```diff
@@ -1094,6 +1106,80 @@ sub upgrade_registry {
     return $self;
 }
 
+sub _find_planned_deployed_divergence_idx {
+    my ($self, $from_idx, @deployed_changes) = @_;
+    my $i = -1;
+    my $plan = $self->plan;
+
+    foreach my $change (@deployed_changes) {
+        $i++;
+        return $i if $i + $from_idx >= $plan->count || $change->script_hash ne $plan->change_at($i + $from_idx)->script_hash;
+    }
+
+    return -1;
+}
+
+sub planned_deployed_common_ancestor_id {
```

Comment:
I didn't see a test for this method, or for `_find_planned_deployed_divergence_idx`. Are they implicit in the tests for `check`?